test/e2e/operator_test.go

```diff
@@ -263,6 +252,30 @@ func setImage(deployment *appsv1.Deployment, image string) {
 	}
 }
 
+func checkCurrentDNSImage(t *testing.T, cl client.Client, expectedImage string) error {
+	err := wait.PollImmediate(1*time.Second, 3*time.Minute, func() (bool, error) {
+		podList := &corev1.PodList{}
+		if err := cl.List(context.TODO(), podList, client.InNamespace("openshift-dns")); err != nil {
+			t.Logf("failed to get pod list in openshift-dns namespace: %v", err)
+			return false, nil
+		}
+
+		for _, pod := range podList.Items {
+			for _, container := range pod.Spec.Containers {
+				if container.Name == "dns" {
+					if container.Image == expectedImage {
+						return true, nil
+					}
+					break
+				}
+			}
+		}
+		return false, nil
```

Comment:
We want to verify that every pod has the expected image, but it looks like this logic only makes sure that some pod does.  We need to invert the logic as follows:  
```suggestion
		for _, pod := range podList.Items {
			for _, container := range pod.Spec.Containers {
				if container.Name == "dns" {
					if container.Image != expectedImage {
						return false, nil
					}
					break
				}
			}
		}
		return true, nil
```

Reply:
huh, nice catch!

Reply:
fixed.